docs/DeveloperGuide.md

```diff
@@ -1,34 +1,284 @@
+bus route.
+
+The class diagram in the figure below shows how different classes used for implementation of the `/routemap` command 
+are linked to each other.
+
+![RouteCommandClass](DG_Diagrams/RouteMapCommandClass.png)
+
+The `RouteMapCommand#executeCommand()` method of RouteMapCommand Class executes the command in the following steps:
+1. Calls `BusData#selectBus()` to find the user-specified bus in the bus data list. If found, the Bus object will be 
+returned. Else, null is returned.
+2. Calls `Ui#printFullRoute()` to display full route of the specified bus.
+
+The following sequence diagram explains the above steps when the user searches for the full route of a bus.
+
+![Overview](DG_Diagrams/RouteMapCommandSeq.png)
+
+
+### 3.3. Favourite command adder (`/addfav` Feature)
+The `AddFavCommand#executeCommand()` method of AddFavCommand Class executes the command in the following steps:
+1. The `RouteParser#executeCommand` method throws an exception if the command to be saved in FavList is missing.
+2. Calls `AddFavCommand#createFav()` to make a new Fav object to be saved in the FavList.
+    - The `AddFavCommand#createFav()` method creates a Fav object with its command saved as its description if 
+    the user left description empty.
+3. Calls `AddFavCommand#addToFavList()` method to add the Fav object created to the Favlist.
+    - The `AddFavCommand#createFav()` method calls the contains method within Favlist to check for any duplicate
+    Fav objects within the list that contains the same command.
+    - If the there are no duplicate Fav objects, Fav object created will be added to the FavList.
+
+`/addfav <description>` is the command that has to be entered by the user to add a previous valid command in to the user's
+list of favourites.
+
+The following sequence diagram illustrates the steps taken by the program when the user calls the `/addfav` command.
+![add favourites](DG_Diagrams/AddFavSequence.png)
+
+### 3.4. Favourite command executor (`/execfav` Feature)
+[Work in Progress]<br>
+`/execfav <INDEX>` is the command which executes a command from the user's list of favourites
+
+The following sequence diagram illustrates the steps taken by the program when the user calls the `/execfav` command.
+![ExecFav_Sequence_Diagram](DG_Diagrams/ExecFavCommand.png)
+
+
+
+### 3.5. Favourite command description modifier (`/descfav` Feature)
+
+`/descfav <index> /to <newDescription>` command allows the user to change the current description of their favourite command
+at location **index** in the list to **newDescription**.
+
+>Note: **index** is the index of the item in the list when the first number is indexed as 1. As a result, 
+>we access this element in the ArrayList using **index** - 1.
+
+The class diagram in the figure below shows how different classes used for implementation of the `/descfav` command 
+are linked to each other. 
+
+![DescFav class diagram](DG_Diagrams/descFavClass.png)
```

Comment:
Good diagram! However you might want to consider if exception throws are necessary for the diagram as too many items in one picture could be confusing.